### ISSUE TITLE
Crash loading terminal program

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/vAmiga#477]
Crashed Thread:        16

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
Assertion failed: (ipl.delayed() == ((iplPipe >> 32) & 0xFF)), function serviceIplEvent, file /Users/hoff/vAmiga/Emulator/Paula/PaulaEvents.cpp, line 44.

**Also, Serial Setting Doesn't save as Default.**

**Comment by bluecursor** [comment]
Latest version, loading the app jrcomm. The issue doesn’t come up when serial loopback is on.

Frank Michielli
Sent with OSX/iOS Spark!

**Comment by dirkwhoffmann** [comment]
This one? 

<img width="904" alt="Bildschirmfoto 2021-03-10 um 18 17 55" src="https://user-images.githubusercontent.com/12561945/110669791-fcd41100-81cc-11eb-82c1-e47d229ea252.png">

**Comment by dirkwhoffmann** [comment]
Is there a way to reproduce the crash deterministically? So far, it didn't crash here.

**Comment by bluecursor** [comment]
Here is a animated gif of what happens + the .adf I use

Frank Michielli
Sent with OSX/iOS Spark!

**Comment by dirkwhoffmann** [comment]
> Here is a animated gif of what happens + the .adf I use

It seems like the attachments didn't go through.

**Comment by bluecursor** [comment]
Frank Michielli
Sent with OSX/iOS Spark!

**Comment by dirkwhoffmann** [comment]
Confirmed. Crashes here, too.

**Comment by dirkwhoffmann** [comment]
The error was in `void Agnus::executeEventsUntil(Cycle cycle)`:

 ```
       ...
       if (isDue<SLOT_IPL>(cycle)) {
            paula.serviceIplEvent();
       }
       if (isDue<SLOT_KBD>(cycle)) {
            keyboard.serviceKeyboardEvent(slot[SLOT_KBD].id);
       }
       if (isDue<SLOT_TXD>(cycle)) {
            uart.serviceTxdEvent(slot[SLOT_TXD].id);
       }
       ...
```

The IPL slot (which handles the CPU interrupt line) was served before the TXD slot (which handles the UART). This is wrong. TXD events need to be served before IPL events. Changing the execution order fixes the issue.

**Comment by dirkwhoffmann** [comment]
Fixed in v1.0b1
